csrc/ops/arith.cpp

```diff
@@ -959,6 +959,52 @@ NVFUSER_DEFINE_INT_ONLY_OP(bitwise_right_shift, Rshift)
 NVFUSER_DEFINE_INT_ONLY_OP(gcd, Gcd)
 #undef NVFUSER_DEFINE_INT_ONLY_OP
 
+// The logical_right_shift operation shifts the value's bits to the right.
+// If the value is negative, it appends zeros to the front of the value.
+// The sign is preserved with arithmetic_right_shift, so ones are pushed to the
+// most significant bit.
+//
+// An alternate approach is to cast the value to an unsigned integer, perform
+// the right shift, and then cast back to the original value. In C++, unsigned
+// integers are shifted with logical right shift.
+template <typename LHS, typename RHS>
+TORCH_CUDA_CU_API typename std::conditional<
+    std::is_same<LHS, TensorView*>::value ||
+        std::is_same<RHS, TensorView*>::value,
+    TensorView*,
+    Val*>::type
+logical_right_shift_helper(LHS x, RHS shift) {
+  auto sizeof_int_dtype = (x->dtype() == PrimDataType::Int) ? 64L : 32L;
+
+  auto neg_one = IrBuilder::create<Val>(x->container(), -1L);
+  auto one = IrBuilder::create<Val>(x->container(), 1L);
+  auto two = IrBuilder::create<Val>(x->container(), 2L);
+  auto num_bits_scalar =
+      IrBuilder::create<Val>(x->container(), sizeof_int_dtype);
+
+  auto mask =
+      where(eq(shift, num_bits_scalar), neg_one, sub(pow(two, shift), one));
```

Comment:
Should this be the following?
```suggestion
      where(ge(shift, num_bits_scalar), neg_one, sub(pow(two, shift), one));
```